platform/ui/src/components/LoadingIndicatorTotalPercent/LoadingIndicatorTotalPercent.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+
+import LoadingIndicatorProgress from '../LoadingIndicatorProgress';
+
+interface Props {
+  className?: string;
+  totalNumbers: number | null;
+  percentComplete: number | null;
+  loadingText?: string;
+  targetText?: string;
+}
+
+/**
+ *  A React component that renders a loading indicator but accepts a totalNumbers
+ * and percentComplete to display a more detailed message.
+ */
+function LoadingIndicatorTotalPercent({
+  className,
+  totalNumbers,
+  percentComplete,
+  loadingText = 'Loading...',
+  targetText = 'segments',
```

Comment:
Why would the default be segments. Is this `LoadingIndicatorTotalPercent` not some generic component that applies to things that might not be segments?

Reply:
I mean it is a interval of something, It can do `Loaded N of M something`, what do you suggest for a default for `something`

Reply:
IMHO, just setting it as the blank string is general enough, or make it a mandatory prop.